sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -379,8 +379,9 @@ private[hive] class HiveClientImpl(
         tableType = h.getTableType match {
           case HiveTableType.EXTERNAL_TABLE => CatalogTableType.EXTERNAL
           case HiveTableType.MANAGED_TABLE => CatalogTableType.MANAGED
-          case HiveTableType.INDEX_TABLE => CatalogTableType.INDEX
           case HiveTableType.VIRTUAL_VIEW => CatalogTableType.VIEW
+          case HiveTableType.INDEX_TABLE =>
+            throw new AnalysisException("Hive index table is not supported.")
```

Comment:
Any function who invokes `getTableOption` might trigger it.

For example, when we create a table that has the same name as an existing Index table, the message we will get is 
```
Hive index table is not supported
```
This might be confusing to external users, because they are not creating a Hive index table. 

In Hive, the error message we will get is
```
hive> create table default__t21_id1__ (col1 int);
FAILED: Execution Error, return code 1 from org.apache.hadoop.hive.ql.exec.DDLTask. AlreadyExistsException(message:Table default__t21_id1__ already exists)
```